processing/src/test/java/org/apache/druid/concurrent/ExecsTest.java

```diff
@@ -118,6 +123,37 @@ public void run()
     producer.shutdown();
   }
 
+  @Test
+  public void testSynchronousQueueSingleThreadedExecutor() throws Exception
+  {
+    // The implementation of Execs.newBlockingSingleThreaded() rejectedExecutionHandler should not add tasks when it's in shutDown state
+    // When capacity is 0, a SynchronousQueue is used and if a task is put in it in ShutDown state, it will forever stuck in WAITING state
+    // as executor will not take() the task to schedule it.
+    final ListeningExecutorService intermediateTempExecutor = MoreExecutors.listeningDecorator(
+        Execs.newBlockingSingleThreaded("[TASK_ID]-appenderator-abandon", 0)
+    );
+    Callable<Void> task = () -> {
+      try {
+        Thread.sleep(500); // Simulate long-running task
+      }
+      catch (InterruptedException e) {
+        Thread.currentThread().interrupt(); // Restore interrupted status
+      }
+      return null;
+    };
+
+    // Submit multiple tasks together
+    ListenableFuture<Void> unused = intermediateTempExecutor.submit(task);
```

Comment:
If you don't plan to use the return value, you need not assign it to anything. Having a variable named `unused` is confusing.

Reply:
I did it like this. Got these errors on tests
```
Error:  COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
Error:  /home/runner/work/druid/druid/processing/src/test/java/org/apache/druid/concurrent/ExecsTest.java:[146,35] error: [CheckReturnValue] The result of `submit(...)` must be used
  If you really don't want to use the result, then assign it to a variable: `var unused = ...`.
  
  If callers of `submit(...)` shouldn't be required to use its result, then annotate it with `@CanIgnoreReturnValue`.
  
    (see https://errorprone.info/bugpattern/CheckReturnValue)
  Did you mean 'var unused = intermediateTempExecutor.submit(task);' or to remove this line?
Error:  /home/runner/work/druid/druid/processing/src/test/java/org/apache/druid/concurrent/ExecsTest.java:[147,35] error: [CheckReturnValue] The result of `submit(...)` must be used
```
Any idea how can I fix this?

Reply:
You could perform an assertion on the return value, like it is non null and the future is not complete yet.